src/executables/csql.c

```diff
@@ -3330,3 +3358,161 @@ get_host_ip (unsigned char *ip_addr)
 
   return 0;
 }
+
+static int
+csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
+{
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char dbname[MAXLOGNAME + CUB_MAXHOSTNAMELEN + 2];
+  char username[DB_MAX_USER_LENGTH + 1];
+  const char *delim = " @\n";
+  char *save_ptr_strtok = NULL;
+  char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
+  char *p = NULL;
+  int error_code;
+  CSQL_ARGUMENT csql_new_arg;
+
+  /*verify @hostname */
+  host_name_ptr = strchr (argument, '@');
+  if (host_name_ptr != NULL)
+    {
+      if (*(host_name_ptr + 1) == '\0')
+	{
+	  host_name_ptr = NULL;
+	  csql_Error_code = CSQL_ERR_CONNECT;
+	  return DO_CMD_FAILURE;
+	}
+    }
+
+  strcpy (dbname, argument);
+  strcpy (username, argument);
+
+  /*find db name following the user name */
+  db_name_ptr = strrchr (dbname, ' ');
+  if (db_name_ptr != NULL)
+    {
+      if (*(db_name_ptr + 1) == '\0')
+	{
+	  db_name_ptr = NULL;
+	  csql_Error_code = CSQL_ERR_CONNECT;
+	  return DO_CMD_FAILURE;
+	}
+      else
+	{
+	  db_name_ptr += 1;
+	}
+    }
+
+  user_name_ptr = strtok_r (username, delim, &save_ptr_strtok);
+  if (user_name_ptr == NULL)
+    {
+      csql_Error_code = CSQL_ERR_CONNECT;
+      return DO_CMD_FAILURE;
+    }
+
+  memset (&csql_new_arg, 0, sizeof (CSQL_ARGUMENT));
+  memcpy (&csql_new_arg, csql_arg, sizeof (CSQL_ARGUMENT));
```

Comment:
You don't need to **memset** (), **memcpy** () will overide it.

Reply:
Right, I worried about any trash value (ex: 0x5348798378)

It is not necessary to use memset(). I will remove it thank you!